charts/scalar-manager/templates/scalar-manager/deployment.yaml

```diff
@@ -0,0 +1,74 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "scalar-manager.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "scalar-manager.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "scalar-manager.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/scalar-manager/configmap.yaml") . | sha256sum }}
+        {{- if .Values.podAnnotations }}
+        {{- toYaml .Values.podAnnotations | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "scalar-manager.selectorLabels" . | nindent 8 }}
+        {{- if .Values.podLabels }}
+        {{- toYaml .Values.podLabels | nindent 8 }}
+        {{- end }}
+    spec:
+      restartPolicy: Always
+      serviceAccountName: {{ include "scalar-manager.serviceAccountName" . }}
+      automountServiceAccountToken: {{ .Values.serviceAccount.automountServiceAccountToken }}
+      containers:
+        - name: {{ .Chart.Name }}-api
+          image: "{{ .Values.api.image.repository }}:{{ .Values.api.image.tag | default .Chart.AppVersion }}"
+          resources:
+            {{- toYaml .Values.api.resources | nindent 12 }}
+          ports:
+            - containerPort: 8080
+          imagePullPolicy: {{ .Values.api.image.pullPolicy }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          volumeMounts:
+            - name: api-application-properties-volume
+              mountPath: /app/application.properties
+              subPath: scalar-manager-api-application.properties
+        - name: {{ .Chart.Name }}-web
+          image: "{{ .Values.web.image.repository }}:{{ .Values.web.image.tag | default .Chart.AppVersion }}"
+          resources:
+            {{- toYaml .Values.web.resources | nindent 12 }}
+          ports:
+            - containerPort: 3000
+          imagePullPolicy: {{ .Values.web.image.pullPolicy }}
+          securityContext:
+            runAsUser: 1000
```

Comment:
## Question 1

```suggestion
            runAsUser: 1001
```
It seems that we use `1001` on the `Dockerfile` side of Scalar Manager API.
https://github.com/scalar-labs/scalar-manager-web/blob/main/Dockerfile#L41

Is this UID `1000` correct? Or, we should use `1001` instead of `1000`?

## Question 2

It seems that we specify the `USER` directive on the `Dockerfile` side, and we set `runAsNonRoot: true` on the helm chart side as a default value.

In this case, I think the `scalar-manager-api` container image will run with UID `1001` automatically.

Do we really need to specify `runAsUser` as a hard-coded value here? If we don't need it, I think it would be better to remove this hard-coded value from here. (For example, I guess this hard-coded value might cause some issues in the OpenShift environment.)

Reply:
@kota2and3kan 
Thank you for your advice.

1000 was a typo. It should be `1001`.

Regarding the question 2.

I did this because I encountered an issue from creating the container with the `runAsNonRoot` security context. This is the error message from `kubectl describe pod <pod-name>`.

```
  Warning  Failed     12s (x7 over 65s)  kubelet            Error: container has runAsNonRoot and image has non-numeric user (nextjs), cannot verify user is non-root (pod: "scalar-manager-57cdcb57d4-pbvvc_scalar-manager(da3940c6-0d80-4287-aa2a-1e9b1f8adc6d)", container: scalar-manager-web)
```

I am not sure if it's because of the USER command in the Dockerfile here https://github.com/scalar-labs/scalar-manager-web/blob/main/Dockerfile#L54 

It specifies the user name but not the user ID.

I tried to change it locally with
```
USER 1001
```

like we did for the API
https://github.com/scalar-labs/scalar-manager-api/blob/main/Dockerfile#L22

However, my internet is too slow here to finish building the image so I couldn't really test it.
(the node packages are just a mess ...

Anyway, since it's not a good practice to use `runAsUser` context, I removed it in https://github.com/scalar-labs/helm-charts/pull/254/commits/5502ca00d60537a0c9e4eb10b6e81c393b47ffb3

I will create a PR in the web repository to specify the user by user ID instead of the user name and try to test it later with better Internet conditions.

Reply:
The PR using the numeric user ID instead of the user name is submitted in
https://github.com/scalar-labs/scalar-manager-web/pull/70

Reply:
I guess the CI failed because of the removal of `runAsUser`.
We can test it after we merge https://github.com/scalar-labs/scalar-manager-web/pull/70 (the workflow will create the snapshot image and we can re-run the action here)

Reply:
@supl 
Thank you for the updates!

As I mentioned on the Scalar Manager Web side, according to the code comment of Kubernetes, it seems that we MUST use `number (UID)` instead of `name (string)` in the container image if we want to use a non-root user.

- https://github.com/kubernetes/kubernetes/blob/v1.30.1/pkg/kubelet/kuberuntime/kuberuntime_container.go#L331
- https://github.com/kubernetes/kubernetes/blob/v1.30.1/pkg/kubelet/kuberuntime/security_context_others.go#L48-L49